app/models/notification.rb

```diff
@@ -32,11 +32,15 @@ class Notification < ApplicationRecord
   end
 
   def parent_event
-    EventStore::Repository.find_event(parent_event_id)
+    @parent_event ||= EventStore::Repository.find_event(parent_event_id)
+  end
+
+  def notification_categories
+    @notification_categories ||= NotificationCategories.new(user)
```

Comment:
The instance of class `NotificationCategories` is owned by the user. So to me it will be a better semantics to remove this method and define a method `User#notification_categories`.

Then wherever needed, call `user.notification_categories`.

In case we work with multiple Notifications in the same context, that will also reduce the number of objects created.

Reply:
Other than that, the code looks much more conscious. Just this one got me totally confused. Also I didn't read the tests. I don't remember if I read them last time.